docs/migration-guides/v2/lib.md

```diff
@@ -0,0 +1,375 @@
+# Migration guide for ably-js v2
+
+Here’s how to migrate from ably-js v1 to v2:
+
+1. [Stop using functionality that v1 deprecated and which v2 removes](#stop-using-v1-deprecated).
+1. [Stop using other functionality that v2 removes](#stop-using-v2-removed).
+1. [Update to v2 and handle its breaking changes](#update-to-v2-and-handle-breaking-changes).
+1. (Optional) [Stop using functionality that v2 deprecates](#stop-using-v2-deprecated).
+1. (Optional) [Take advantage of new features that v2 introduces](#use-v2-new-features).
+
```

Comment:
```suggestion
## NB: changes to realtime connection sequence and transport order

In previous versions of ably-js, realtime clients with multiple available transports would initially attempt to connect with the transport most likely to succeed (`xhr_polling` in web browsers) and only once the initially connection was made, would we then attempt to 'upgrade' to a preferable transport such as `web_socket`.

In version 2 of the library, realtime clients will instead connect with `web_socket` first and only failover to alternative transports if the initial WebSocket connection attempt fails. For the vast majority of users this will result in a smoother initial connection sequence, however in environment where WebSocket connections are unavailable and time out instead of failing immediately (such as when using a corporate proxy which strips WebSocket headers) this may result in a slower initial connection. Once a connection is first established, transport preference will be cached in web browser local storage so subsequent connections will use the best available transport. If you expect web socket attempts to always fail in your enviornment, you can skip the WebSocket connection step by explicitly providing a list of transports which omits the `web_socket` transport via client options.
```

Reply:
as mentioned on slack, we should also mention that when using `web_socket` as a transport, version 2 of the realtime client will make use of the `wss://ws-up.ably-realtime.com` endpoint, so if you're using a firewall you'll need to ensure that this endpoint is reachable. we can link out to the documentation at https://faqs.ably.com/if-i-need-to-whitelist-ablys-servers-from-a-firewall-which-ports-ips-and/or-domains-should-i-add.

Reply:
Thanks, have very lightly edited the above and added it, and also added a section about whitelisting the endpoint.

Reply:
Resolving since Owen’s away.